packages/xchain-cosmos/src/client.ts

```diff
@@ -282,9 +277,9 @@ class Client implements CosmosClient, XChainClient {
       this.registerCodecs()
 
       const mainAsset = this.getMainAsset()
-      const txHistory = await this.sdkClient.searchTx({
+      const txHistory = await this.getSDKClient().searchTx({
         messageAction,
-        messageSender,
+        messageSender: params?.address,
```

Comment:
Why not `messageSender: (params && params.address) || this.getAddress()` as we had it before?

Reply:
re-added